test-suite-data.json

```diff
@@ -466,5 +466,65 @@
     "qualifiers": null,
     "subpath": null,
     "is_invalid": true
+  },
+  {
+    "description": "a name and version are always required",
+    "purl": "pkg:pub",
+    "canonical_purl": "pkg:pub",
+    "type": "pub",
+    "namespace": null,
+    "name": null,
+    "version": null,
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": true
+  },
+  {
+    "description": "valid base pub purl",
+    "purl": "pkg:pub",
+    "canonical_purl": "pkg:pub/utils@0.1.0",
+    "type": "pub",
+    "namespace": null,
+    "name": "utils",
+    "version": "0.1.0",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "valid hosted pub purl",
+    "purl": "pkg:pub",
+    "canonical_purl": "pkg:pub/utils@0.1.0?hosted_url=https://pub.hosted.org",
```

Comment:
This and the examples below should be updated to the current percent-encoding rules, i.e. `/` must be escaped as `%2F` in the value of the qualifiers in the `canonical_purl`.